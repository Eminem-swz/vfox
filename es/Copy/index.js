import { defineComponent, openBlock, createBlock, createVNode, renderSlot, createTextVNode } from 'vue';

var script = defineComponent({
    name: 'fx-copy',
    props: {
        // 需要复制的文本
        text: {
            type: String,
            required: true
        }
    },
    methods: {
        /**
         * 复制
         */
        onCopy(e) {
            const $el = this.$el;
            try {
                $el.firstElementChild.select();
                document.execCommand('Copy');
                this.$emit('success', {
                    text: $el.firstElementChild.value
                });
            }
            catch (error) {
                this.$emit('error', error);
            }
            this.$emit(e.type, e);
        }
    }
});

const _hoisted_1 = { class: "fx-copy_box" };
const _hoisted_2 = /*#__PURE__*/createTextVNode("复制");

function render(_ctx, _cache, $props, $setup, $data, $options) {
  return (openBlock(), createBlock("div", {
    class: "fx-copy",
    onClick: _cache[1] || (_cache[1] = (...args) => (_ctx.onCopy && _ctx.onCopy(...args)))
  }, [
    createVNode("input", {
      type: "text",
      value: _ctx.text,
      class: "fx-copy_input"
    }, null, 8 /* PROPS */, ["value"]),
    createVNode("div", _hoisted_1, [
      renderSlot(_ctx.$slots, "default", {}, () => [
        _hoisted_2
      ])
    ])
  ]))
}

script.render = render;
script.__file = "src/Copy/Copy.vue";

const _Copy = Object.assign(script, {
    install: function (app) {
        app.component(script.name, script);
    }
});

export default _Copy;
