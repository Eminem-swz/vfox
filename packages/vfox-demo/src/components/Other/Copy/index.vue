<template>
  <fx-group title="基本用法">
    <div class="exp-copy-pad">
      <fx-copy class="exp-copy-box" text="复制的文本">
        <fx-button type="primary">点击复制</fx-button>
      </fx-copy>
    </div>
  </fx-group>
  <fx-group title="事件监听">
    <div class="exp-copy-pad">
      <fx-copy
        class="exp-copy-box"
        text="复制的文本2"
        @success="onSuccess"
        @error="onError"
      >
        <fx-button type="primary">点击复制</fx-button>
      </fx-copy>
    </div>
  </fx-group>
</template>

<script lang="ts">
import { showToast } from '@/index'
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'ExpCopy',
  setup() {
    const onSuccess = (text: string) => {
      showToast({
        title: `${text}`,
        type: 'success'
      })
    }

    const onError = (error: Error) => {
      showToast({
        title: error.message,
        type: 'fail'
      })
    }

    return {
      onSuccess,
      onError
    }
  }
})
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style lang="scss">
.exp-copy {
  &-box {
    display: flex !important;
    flex: 1;
  }

  &-pad {
    padding: 12px 16px;
  }
}
</style>
