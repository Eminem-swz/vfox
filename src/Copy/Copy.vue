<template>
  <div class="fx-copy" @click="onCopy">
    <input type="text" :value="text" class="fx-copy_input" ref="inputEl" />
    <div class="fx-copy_box">
      <slot>{{ locale.copyText }}</slot>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref } from 'vue'
import { locale } from '@/Locale'

export default defineComponent({
  name: 'fx-copy',
  props: {
    // 需要复制的文本
    text: {
      type: String,
      required: true
    }
  },
  emits: {
    success: (payload: string) => typeof payload === 'string',
    error: (e: Error) => e instanceof Error
  },
  setup(props, { emit }) {
    const inputEl = ref<HTMLInputElement>()

    function onCopy() {
      try {
        const $el = inputEl.value as HTMLInputElement

        $el.select()
        document.execCommand('Copy')

        emit('success', $el.value ?? '')
      } catch (error) {
        emit('error', error as Error)
      }
    }

    return {
      inputEl,
      onCopy,
      locale
    }
  }
})
</script>
