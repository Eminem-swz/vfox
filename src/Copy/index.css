.fx-copy {
  display: inline-block;
  position: relative;
}

.fx-copy_input {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  padding: 0;
  border: none;
  opacity: 0;
  z-index: 0;
}

.fx-copy_box {
  position: relative;
  width: 100%;
  height: 100%;
  display: flex;
}
