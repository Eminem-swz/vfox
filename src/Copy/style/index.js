import '../../style/common'
import '../index.scss'
