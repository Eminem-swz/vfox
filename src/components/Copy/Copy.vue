<template>
  <div :class="[prefix + '-copy']" @click="onCopy">
    <input type="text" :value="text" :class="[prefix + '-copy_input']" />
    <div :class="[prefix + '-copy_box']"><slot>复制</slot></div>
  </div>
</template>

<script>
import { SDKKey } from '../../config'

export default {
  name: SDKKey + '-copy',
  props: {
    // 需要复制的文本
    text: {
      type: String,
      required: true
    }
  },
  data() {
    return { prefix: SDKKey }
  },
  methods: {
    /**
     * 复制
     */
    onCopy(e) {
      const $el = this.$el

      try {
        $el.firstElementChild.select()
        document.execCommand('Copy')

        this.$emit('success', {
          text: $el.firstElementChild.value
        })
      } catch (error) {
        this.$emit('error', error)
      }

      this.$emit(e.type, e)
    }
  }
}
</script>

<style lang="scss">
@import '../component.module.scss';

.#{$prefix}-copy {
  display: inline-block;
  position: relative;

  &_input {
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    padding: 0;
    border: none;
    opacity: 0;
    z-index: 0;
  }

  &_box {
    position: relative;
    display: block;
    width: 100%;
    height: 100%;
    display: flex;
  }
}
</style>
