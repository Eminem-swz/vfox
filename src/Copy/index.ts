import { withInstall } from '@/helpers/with-install'
import Copy from './Copy.vue'

export { Copy }
export default withInstall(Copy)
