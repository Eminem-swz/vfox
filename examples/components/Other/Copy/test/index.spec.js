/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import Copy from '@/components/Copy'

describe('Copy', () => {
  test('should render prop text="content" correctly', () => {
    const wrapper = mount(Copy, {
      propsData: {
        text: 'content'
      },
      slots: {
        default: 'content'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
