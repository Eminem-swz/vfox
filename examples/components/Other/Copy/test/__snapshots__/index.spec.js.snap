// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy should render prop text="content" correctly 1`] = `
<div class="fx-copy"><input type="text" class="fx-copy_input">
  <div class="fx-copy_box">content</div>
</div>
`;
